lib/project_types/script/layers/infrastructure/script_service.rb

```diff
@@ -34,11 +34,13 @@ def set_app_script(
             configurationUi: script_config.configuration_ui,
             configurationDefinition: script_config.configuration&.to_json,
             moduleUploadUrl: module_upload_url,
-            library: {
-              language: library[:language],
-              version: library[:version],
-            },
           }
+
+          variables.merge!({
+            language: library[:language],
+            version: library[:version],
+          }) if library
```

Comment:
This allows the CLI to provide a `nil` library object. This is normal for Wasm projects since we don't have a library.

Reply:
IIRC, there was a decision that the Partners dashboard would show the protocol version instead of the library version. If that's the case, is the plan to update Script Service later to support a `protocol_version` instead of this?

Reply:
I'm not sure if we ever committed to that decision. In any case, we already send Script Service the protocol version via 
 `schemaMajorVersion` in the [GQL mutation](https://github.com/Shopify/shopify-cli/blob/f017613b4ee39d4a0766f4b85801cb85a3e92592/lib/project_types/script/layers/infrastructure/script_service.rb#L31)